packages/trader/src/Modules/Contract/Components/InfoBox/info-box-longcode.tsx

```diff
@@ -1,41 +1,66 @@
 import classNames from 'classnames';
 import React from 'react';
-import { Icon, Text } from '@deriv/components';
-import { localize } from '@deriv/translations';
-import { isMobile, TContractInfo } from '@deriv/shared';
+import { Icon, Text, Modal, Button } from '@deriv/components';
+import { Localize } from '@deriv/translations';
+import { TContractInfo } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
 
 type TInfoBoxLongcode = { contract_info: TContractInfo };
 
-const InfoBoxLongcode = ({ contract_info }: TInfoBoxLongcode) => {
-    const max_longcode_length = isMobile() ? 47 : 150;
+const InfoBoxLongcode = observer(({ contract_info }: TInfoBoxLongcode) => {
+    const {
+        ui: { is_mobile },
+    } = useStore();
+    const max_longcode_length = is_mobile ? 47 : 150;
     const [is_collapsed, setIsCollapsed] = React.useState(true);
 
-    const handleToggle = (ev: React.MouseEvent<HTMLElement>) => {
-        ev.preventDefault();
+    const handleToggle = () => {
         setIsCollapsed(!is_collapsed);
     };
 
     return (
-        <div className='info-box-longcode'>
+        <div className='info-box-longcode' onClick={is_mobile ? handleToggle : undefined} onKeyDown={handleToggle}>
```

Comment:
Because of adding `onKeyDown` I can open modal by pressing on any button, not only 'Enter'. Is it possible to add check (like `if(e.key === 'Enter')`)?
I guess another way is to remove `onClick` and `onKeyDown` from container and increase the clickable area on mobile  of 'View more'. It might be done with the usage of pseudo elements like:
```
.info-box-longcode-text:before {
  content: '';
  position: absolute;
  width: some width;
  height: some height;
  top: some positioning ;
  left: some positioning;
}
```

Reply:
why do we want keydown ?

Reply:
I just added this to get rid of solarcloud bug since it doesn't allow adding onclick without any keyboard event. I tried it and I think we don't neet to restrict it to any key as it works pretty well this way.

Reply:
I don't want to increase  the clickable area though, the longcode and the button itself have variable sizes and it will probably break at some pointm making more or less area clickable 😄